thirdparty/sol2-2.20.0/sol.hpp

```diff
@@ -1719,7 +1719,8 @@ namespace sol {
 #else
 			is_specialization_of<meta::unqualified_t<T>, basic_string_view>,
 #endif

```

Comment:
as with pybind11, could we fetch the sources configuration time?

Reply:
I'd rather not change it now, in the context of this PR. Maybe on the next upgrade?

Reply:
I am OK with doing it in another PR, I am also OK with not making the change if this is a rarely touched dependency (it seems to be), although I feel that having a separate patch file helps with upgrades/tracking changes

Reply:
I agree that it would make the changes more visible, although I tried my best by adding comments where I patched the source.
Changing this to ExternalProject would be less than ideal, because the header would not be available before the first build.
Because of this I would go for FetchContent, but due to its different syntax, I would most likely need to manually invoke the patch command at the right times instead of letting cmake take care of it. This is somewhat unknown territory and I would need to do more research, that's why I prefer not to do it now.

Reply:
makes sense